trie/patriciaMerkleTrie.go

```diff
@@ -217,6 +237,11 @@ func (tr *patriciaMerkleTrie) RootHash() ([]byte, error) {
 	tr.mutOperation.Lock()
 	defer tr.mutOperation.Unlock()
 
+	err := tr.updateTrie()
```

Comment:
can RootHash be called also from an API for the current trie (the one being updated)?

If yes then probably here will be some degrade as it will generate also the write in trie, before computing the root hash

Reply:
No, the API uses another trie, not the one that is used for processing.